tests/executor/test_executor.py

```diff
@@ -188,3 +198,146 @@ def test_clear(tmpfile):
         assert int(status.tags['index_size']) == 0
+
+
+@patch.dict(os.environ, {'JINA_AUTH_TOKEN': ''})
+def test_remote_storage(tmpfile):
+    os.environ['JINA_AUTH_TOKEN'] = 'ed17d158d95d3f53f60eed445d783c80'
+    clear_hubble()
+
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        f.post(on='/index', inputs=docs)
+        time.sleep(2)
+        f.post(on='/backup', parameters={'target_name': 'backup_docs'})
+        time.sleep(2)
+
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={'n_dim': D, 'restore_key': 'backup_docs'},
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        status = f.post(on='/status', return_results=True)[0]
+
+    assert int(status.tags['total_docs']) == N
+    assert int(status.tags['index_size']) == N
+
+
+def test_local_storage(tmpfile):
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        f.post(on='/index', inputs=docs)
+        time.sleep(2)
+        f.post(on='/backup')
+        time.sleep(2)
+
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={'n_dim': D},
+        workspace=tmpfile,
+        shards=1,
+    )
+    with f:
+        status = f.post(on='/status', return_results=True)[0]
+
+    assert int(status.tags['total_docs']) == N
+    assert int(status.tags['index_size']) == N
+
+
+@patch.dict(os.environ, {'JINA_AUTH_TOKEN': ''})
+def test_remote_storage_with_shards(tmpfile):
+    os.environ['JINA_AUTH_TOKEN'] = 'ed17d158d95d3f53f60eed445d783c80'
+    docs = gen_docs(N)
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+        },
+        workspace=tmpfile,
+        shards=3,
+        polling={'/index': 'ANY', '/search': 'ALL', '/backup': 'ALL', '/status': 'ALL'},
+    )
+    with f:
+        f.post(on='/index', inputs=docs)
+        time.sleep(2)
+        f.post(
+            on='/backup',
+            parameters={'target_name': 'backup_docs_with_shards'},
+        )
+        time.sleep(2)
+
+    f = Flow().add(
+        uses=AnnLiteIndexer,
+        uses_with={
+            'n_dim': D,
+            'restore_key': 'backup_docs_with_shards',
+        },
+        workspace=tmpfile,
+        shards=3,
+        polling={'/index': 'ANY', '/search': 'ALL', '/backup': 'ALL', '/status': 'ALL'},
+    )
+    with f:
+        status = f.post(on='/status', return_results=True)
+
+    total_docs = 0
+    index_size = 0
+    for stat in status:
+        total_docs += stat.tags['total_docs']
+        index_size += stat.tags['index_size']
+    assert total_docs == N
+    assert index_size == N
+    clear_hubble()
+
+
+def test_local_storage_with_shards(tmpfile):
```

Comment:
We can use a parameter to specify the number of shards and we don't need to separate unit tests anymore.